it/test/controllers/NirmsQuestionControllerISpec.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import helpers.ItTestBase
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.{WSClient, WSRequest}
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+
+class NirmsQuestionControllerISpec extends ItTestBase {
+
+  lazy val client: WSClient = app.injector.instanceOf[WSClient]
+
+  private val url = s"http://localhost:$port${routes.NirmsQuestionController.onPageLoad.url}"
```

Comment:
It is relying on the routes not the controller, so I'm not too worried about it because we aren't trying to test the routes have been setup correctly here? 

But I have no strong opinions, we switched to this way after another code review comment :D

Reply:
leave it i'm just being picky :p

Reply:
@besscerule  The url needs to be correct for the test to call the right endpoint, not hardcoding it and instead setting it to the routes url means we won't have to update this test if we change the path